### PR TITLE
Samtools flagstats

### PR DESCRIPTION
**Pull request by emmarousseau**
## Description

Samtools flagstats

## Checklist before requesting a review

- [x] I have performed a self-review of my code

- [x] Conforms to the [Contributing guidelines](https://github.com/viash-hub/base/blob/main/CONTRIBUTING.md)

- [x] Proposed changes are described in the [CHANGELOG.md](https://github.com/viash-hub/base/blob/main/CHANGELOG.md)

- [x] I have tested my code with `viash ns test --parallel -q <name or namespace>`

- Check the correct box. Does this PR contain:
  - [ ] Breaking changes
  - [x] New functionality
  - [ ] Major changes
  - [ ] Minor changes
  - [ ] Documentation
  - [ ] Bug fixes

### CODE REVIEW

File: src/samtools/flagstat/config.vsh.yaml
Comment (rcannood) [inline src/samtools/flagstat/config.vsh.yaml:1]: Could you move this component to `src/samtools/samtools_flagstat`? That is, `src/<namespace>/<name>`.

File: src/samtools/flagstat/config.vsh.yaml [hunk 1]
```diff
@@ -0,0 +1,50 @@
+name: samtools_flagstat
+namespace: samtools
+description: Counts the number of alignments in SAM/BAM/CRAM files for each FLAG type.
+keywords: [ stats, mapping, counts, bam, sam, cram ]
+links:
+  homepage: https://www.htslib.org/
+  documentation: https://www.htslib.org/doc/samtools-flagstat.html
+  repository: https://github.com/samtools/samtools.git
+references: 
+  doi: 10.1093/bioinformatics/btp352, 10.1093/gigascience/giab008
```
Comment (rcannood) [inline src/samtools/flagstat/config.vsh.yaml:10]:
```suggestion
  doi: [10.1093/bioinformatics/btp352, 10.1093/gigascience/giab008]
```

Or 

```suggestion
  doi:
    - 10.1093/bioinformatics/btp352
    - 10.1093/gigascience/giab008

File: src/samtools/flagstat/config.vsh.yaml [hunk 2]
```diff
@@ -0,0 +1,50 @@
+name: samtools_flagstat
+namespace: samtools
+description: Counts the number of alignments in SAM/BAM/CRAM files for each FLAG type.
+keywords: [ stats, mapping, counts, bam, sam, cram ]
+links:
+  homepage: https://www.htslib.org/
+  documentation: https://www.htslib.org/doc/samtools-flagstat.html
+  repository: https://github.com/samtools/samtools.git
```
Comment (rcannood) [inline src/samtools/flagstat/config.vsh.yaml:8]: You can omit the `.git` →

```suggestion
  repository: https://github.com/samtools/samtools
```

File: src/samtools/flagstat/script.sh [hunk 3]
```diff
@@ -0,0 +1,13 @@
...
+## VIASH START
+## VIASH END
+
+set -e
+
+output_file="$par_output"
+
+samtools flagstat \
+    "$par_bam" \
+    > "$output_file"
```
Comment (rcannood) [inline src/samtools/flagstat/script.sh:12]: Is there a reason for the `output_file` environment variable?

```suggestion
samtools flagstat "$par_bam" > "$par_output"
```

**Review by rcannood — COMMENTED** [review]
Thanks for the PR, @emmarousseau! 

I left some minor comments but other than that, everything looks great!

File: src/samtools/flagstat/config.vsh.yaml [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+    path: test.sh
+  - type: file
+    path: test_data
+engines:
+  - type: docker
+    image: quay.io/biocontainers/samtools:1.19.2--h50ea8bc_1
+    setup:
+      - type: docker
+        run: |
+          echo "$(samtools --version 2>&1)" > /var/software_versions.txt
```
Comment (rcannood) [inline src/samtools/flagstat/config.vsh.yaml:47]: This command outputs way too much information; I expect only to see `samtools: "0.0.0"` in `/var/software_versions.txt`. Something like:

```suggestion
          samtools --version 2>&1 | grep -E '^(samtools|Using htslib)' | sed 's#Using ##;s# \([0-9\.]*\)$#: \1#' > /var/software_versions.txt
```

File: src/samtools/flagstat/script.sh (hunk 3 above)
Comment (emmarousseau) [reply src/samtools/flagstat/script.sh:12]: No actually no reason, I forgot to remove it, thanks!

**Review by rcannood — APPROVED** [review]
LGTM! Thanks @emmarousseau!

**Pull request merged**